typing/btype.ml

```diff
@@ -260,7 +260,10 @@ let fold_row f init row =
       (row_fields row)
   in
   match get_desc (row_more row) with
-  | Tvar _ | Tunivar _ | Tsubst _ | Tconstr _ | Tnil | Tof_kind _ ->
+  | Tvar _ | Tunivar _ | Tsubst _ | Tconstr _ | Tnil
+    (* Tof_kind can appear in [row_more] in case the row's row variable was existentially
+       quantified in a GADT *)
```

Comment:
This is helpful, but still doesn't have all of the detail I would want. So I have made #3934 so that there is a good place to put those details.

I *think* that any `Tof_kind` in a row variable will always be of kind `value`, because we are not yet supporting having an open polymorphic variant whose fields are limited to those that mode-cross a certain way.

Reply:
Let's not hold this up on resolving this.